bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/DateTimeTriggerHandler.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.internal.module.handler;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.ModuleHandlerCallback;
+import org.openhab.core.automation.Trigger;
+import org.openhab.core.automation.handler.BaseTriggerModuleHandler;
+import org.openhab.core.automation.handler.TimeBasedTriggerHandler;
+import org.openhab.core.automation.handler.TriggerHandlerCallback;
+import org.openhab.core.config.core.ConfigParser;
+import org.openhab.core.events.Event;
+import org.openhab.core.events.EventFilter;
+import org.openhab.core.events.EventSubscriber;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.events.ItemCommandEvent;
+import org.openhab.core.items.events.ItemEvent;
+import org.openhab.core.items.events.ItemStateEvent;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.scheduler.CronAdjuster;
+import org.openhab.core.scheduler.CronScheduler;
+import org.openhab.core.scheduler.ScheduledCompletableFuture;
+import org.openhab.core.scheduler.SchedulerRunnable;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an ModuleHandler implementation for Triggers which trigger the rule
+ * based on a {@link org.openhab.core.library.types.DateTimeType} stored in an item
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class DateTimeTriggerHandler extends BaseTriggerModuleHandler
+        implements SchedulerRunnable, TimeBasedTriggerHandler, EventSubscriber, EventFilter {
+
+    public static final String MODULE_TYPE_ID = "timer.DateTimeTrigger";
+    public static final String CONFIG_ITEM_NAME = "itemName";
+
+    private static final DateTimeFormatter CRON_FORMATTER = DateTimeFormatter.ofPattern("s m H d M * uuuu");
+
+    private final Logger logger = LoggerFactory.getLogger(DateTimeTriggerHandler.class);
+
+    private final CronScheduler scheduler;
+    private final String itemName;
+    private String cronExpression = "@reboot";
```

Comment:
This magic string `@reboot` looks a bit weird to me. How about simply setting it to `null`, if no cron expression is set yet? Or if you really prefer to set a magic string in order to avoid null values, please at least make it a constant.

Reply:
I can make it a constant. I used this string because it is properly handled in the CronAdjuster, so probably we should make it a constant there. WDYT?

Reply:
Ah, I didn't even know that this value was defined in the CronAdjuster. So yeah, it makes a lot of sense to declare it as a constant in there and just use it.